### PR TITLE
[SYCL]Support LLVM FP intrinsic in llvm-spirv and enable the corresponding …

### PR DESCRIPTION
**Pull request by jinge90**
…builtin in FE

Signed-off-by: gejin <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @andykaylor  @bader  @vzakhari 
This PR includes:
1. Add llvm float point intrinsic support in llvm-spirv
2. Enable corresponding builtin in SemaSYCL
This patch is ported from Andy's work and should fix std::complex issue on Windows platform.
Thanks very much.

**Comment by bader** [comment]
@jinge90, please, commit llvm-spirv translator changes here: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/. They will be pulled to this repository.

**Comment by jinge90** [comment]
> @jinge90, please, commit llvm-spirv translator changes here: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/. They will be pulled to this repository.

Hi, @bader 
Should I close this PR and commit the llvm-spirv translator change into https://github.com/KhronosGroup/SPIRV-LLVM-Translator/ firstly, then create a new PR in intel/llvm to enable the corresponding builtin when the llvm-spirv change is pulled into intel/llvm?
Thanks very much.

**Comment by bader** [comment]
No. Clang and llvm-spriv parts are independent, so just move llvm-spirv patch from this PR to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/.

**Comment by jinge90** [comment]
> No. Clang and llvm-spriv parts are independent, so just move llvm-spirv patch from this PR to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/.

OK,I will move the llvm-spirv part to another patch,commit to SPIRV-LLVM-Translator repo and only keep clang change in this PR.
Thanks very much!

**Review by andykaylor — DISMISSED** [review]
This looks good to me. I think Alexey's suggestion to move the supported builtins to a separate test makes sense.

File: clang/test/SemaSYCL/unsupported_math.cpp
Comment (jinge90) [reply clang/test/SemaSYCL/unsupported_math.cpp]: Moved the "supported" math builtin/function into a new case "supported_math.cpp".

**Comment by jinge90** [comment]
Hi, @andykaylor 
I updated the patch to move supported builtins to a separate test but missed your previous approval, could you help review again?
Thanks very much.

**Comment by vzakhari** [comment]
This looks good to me, but, formally, clang code owners should review.

**Review by elizabethandrews — APPROVED** [review]
LGTM

**Pull request merged**
